### DISCUSSION TITLE
Restrict access to user profile?

### DISCUSSION THREAD

**Discussion by ARCHES-Academy** [statamic/cms#4592] [Q&A]
Is there a way to restrict the users from accessing their own user profile? I've already removed the permission to view users, but their own user is different, of course. However, there's some data that I can't have the users changing.

Can I make it so the user profile link doesn't show? Maybe a custom permission, or something like that?

**Comment by duncanmcclean** [comment]
I'm guessing this is what you're referring to?

![image](https://user-images.githubusercontent.com/19637309/139531185-8c7788de-e413-4771-a7d8-2ce87a8553c5.png)

If so, I'd probably hide the link to it with some JavaScript, like this...

```js
document.querySelector('.global-header .popover-content a[href="https://cms-sandbox.test/cp/account"]').style.display = 'none'
```

**Reply by ARCHES-Academy** [reply]
Yeah, that could work. At the moment I found a workaround that I'm using, but it's probably not the most elegant solution.

I only needed to hid it for one user group, who also have all unique views that the other user groups won't need to access, so for those views I extended the `outside` layout instead of the `layout` layout. I'm not sure what this will change besides the lack of global header & nav bar, but it did remove those two things. 

At that point, I added the global header component to the top again, tweaking a couple things to make it look seamless, and then removed the option (profile) that I didn't want. This made the code slightly messier, but it resulted in the end result I was looking for.

If I hide the option with CSS/Javascript, it's still in the DOM, and therefore it's possible for the tech-savvy user to find and click that link - which should be off-limits. This way, it's never even rendered, and therefore much more secure - I think.
